lapce-proxy/src/terminal.rs

```diff
@@ -307,3 +331,52 @@ fn set_locale_environment() {
         .replace('-', "_");
     std::env::set_var("LC_ALL", locale + ".UTF-8");
 }
+
+#[cfg(not(target_os = "linux"))]
+fn flatpak_get_default_host_shell() -> String {
+    panic!(
+        "This should never be reached. If it is, ensure you don't have a file
+        called .flatpak-info in your root directory"
+    );
+}
+
+#[cfg(target_os = "linux")]
+fn flatpak_get_default_host_shell() -> String {
+    use std::process::Command;
+
+    let env_string = Command::new("flatpak-spawn")
+        .arg("--host")
+        .arg("printenv")
+        .output()
+        .unwrap()
+        .stdout;
+
+    let env_string = String::from_utf8(env_string).unwrap();
+
+    for env_pair in env_string.split('\n') {
+        let name_value: Vec<&str> = env_pair.split('=').collect();
+
+        if name_value[0] == "SHELL" {
+            return name_value[1].to_string();
+        }
+    }
+
+    // In case SHELL isn't set for whatever reason, fall back to this
+    "/bin/sh".to_string()
+}
+
+#[cfg(not(target_os = "linux"))]
+fn is_inside_flatpak() -> bool {
+    false // Flatpak is only available on Linux
+}
+
+#[cfg(target_os = "linux")]
+fn is_inside_flatpak() -> bool {
+    use std::path::Path;
```

Comment:
You can use fully qualified path instead of importing it if you use it just once

```rs
std::path::Path::new
```

Reply:
Personally, I think it's easier to read with the import done separately. But, if this is preferred I can change it.